File: app/src/main/java/com/github/premnirmal/ticker/network/BaseConverter.java
package com.github.premnirmal.ticker.network;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.io.IOException;
import java.io.InputStream;

import retrofit.converter.GsonConverter;
import retrofit.mime.TypedOutput;

/**
 * Created by premnirmal on 4/10/15.
 */
abstract class BaseConverter implements retrofit.converter.Converter {

    protected final Gson gson = new GsonBuilder().create();

    protected String getString(InputStream is) throws IOException {
        int ch;
        StringBuilder sb = new StringBuilder();
        while ((ch = is.read()) != -1)
            sb.append((char) ch);
        return sb.toString();
    }

    @Override
    public TypedOutput toBody(Object object) {
        return new GsonConverter(gson).toBody(object);
    }
}

File: app/src/main/java/com/github/premnirmal/ticker/network/BaseConverter.kt
package com.github.premnirmal.ticker.network

import com.google.gson.Gson
import okhttp3.ResponseBody
import retrofit2.Converter
import java.io.IOException
import java.io.InputStream

/**
 * Created by premnirmal on 3/3/16.
 */
internal abstract class BaseConverter<T>(val gson: Gson) : Converter<ResponseBody, T> {

  @Throws(IOException::class) protected fun getString(`is`: InputStream): String {
    var ch: Int
    val sb = StringBuilder()
    do {
      ch = `is`.read()
      sb.append(ch.toChar())
    } while (ch != -1)
    return sb.toString()
  }
}